### DISCUSSION TITLE
vue-query testing

### DISCUSSION THREAD

**Discussion by aabeborn** [TanStack/query#4479] [Q&A]
Hi I'm currently testing a vue component using `vue-test-utils`, but mounting the component, which has `useInfiniteQuery` inside of it the following error appears:
`Error: No 'queryClient' found in Vue context, use 'VueQueryPlugin' to properly initialize the library.`

This is strange because I've added the plugin to the mounting method as follows:
```
const wrapper = shallowMount(ProductsList, {
			global: {
				plugins: [VueQueryPlugin],
			},
		});
```

Any idea? The only solution I found is to mock the package and `useInfiniteQuery`, but working with reactive mocks isn't wonderful.
Thank you!

**Comment by max-zinn** [comment]
Hi, have you tried creating a localVue? I am using vue 2.6 and the following is working for us:

```typescript
import { createLocalVue } from "@vue/test-utils";
import { VueQueryPlugin } from "@tanstack/vue-query";
import { shallowMount } from "@vue/test-utils";

const localVue = createLocalVue();
localVue.use(VueQueryPlugin);

shallowMount(component, {
  localVue,
});
```

**Reply by onekiloparsec** [reply]
`createLocalVue` doesn't exist anymore in Vue3. 

I am using the same kind of code as @aabeborn but to no avail. Testing page of vue-quersy is [empty](https://tanstack.com/query/latest/docs/vue/guides/testing), what a pity.

**Reply by marianocodes** [reply]
@onekiloparsec  did you find a solution? testing page is still empty

**Reply by horikx** [reply]
> `createLocalVue` doesn't exist anymore in Vue3.
> 
> I am using the same kind of code as @aabeborn but to no avail. Testing page of vue-quersy is [empty](https://tanstack.com/query/latest/docs/vue/guides/testing), what a pity.

You can install the plugin in config.global.plugins and the error goes away
